PARTITION_PINNING.md

```diff
@@ -0,0 +1,136 @@
+# Partition Pinning Feature
+
+This document explains how to use the partition pinning feature to improve benchmark performance by having each client instance connect to specific partitions.
+
+## Overview
+
+By default, the benchmark client connects to all partitions across all brokers in the Camunda 8 cluster. The partition pinning feature allows each benchmark client to be "pinned" to a specific subset of partitions, ideally located on the same broker, which can significantly improve performance.
+
+## How It Works
+
+1. **Client Identity**: Each client has a unique numerical identity (pod ID from Kubernetes StatefulSet)
+2. **Message-based Process Starting**: Instead of creating process instances directly, the client publishes messages with correlation keys that route to specific partitions
+3. **Partition-specific Job Types**: Job workers only subscribe to job types that include their pod ID, ensuring they only handle jobs from their assigned partitions
+
+## Configuration
+
+### Required Properties
+
+```properties
+# Enable partition pinning
+benchmark.client.enable-partition-pinning=true
+
+# Pod ID (set automatically from Kubernetes pod metadata)
+benchmark.client.pod-id=${POD_NAME}
+
+# Total number of partitions in your Zeebe cluster
+benchmark.client.partitionCount=8
```

Comment:
Let's use 9 partitions in the example. For the round robin to be balanced, we should recommend the number of partitions to be divisible by the number benchmark replicas.